### PR TITLE
Suggested code cleanups

### PR DESCRIPTION
**Pull request by SeanFeldman**
Mostly things like
- unused namespace
- exception re-throwing
- auto properties / expression based properties
- redundant else/ default case
- redundant parenthesis 
- redundant variable initialization
- unnecessary await
- constructor initialization
- //??? comments where logic is in question
- null-conditional operator
- method visibility
- nameof() instead of hardcoding argument name
- Assert.Throws instead of try/catch

### CODE REVIEW

File: src/Microsoft.Azure.EventHubs/Amqp/AmqpMessageConverter.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public static AmqpMessage EventDatasToAmqpMessage(IEnumerable<EventData> eventDa
         {
             AmqpMessage returnMessage = null;
             int dataCount = eventDatas.Count();
```
Comment (serkantkaraca) [inline src/Microsoft.Azure.EventHubs/Amqp/AmqpMessageConverter.cs]: >dataCount [](start = 16, length = 9)

Good catch. We should just check the dataCount here. eventDatas should never be null. Please throw if it is.

File: src/Microsoft.Azure.EventHubs/Amqp/Management/RealProxy.cs [hunk 2]
```diff
@@ -449,7 +451,7 @@ static void Convert(ILGenerator il, Type source, Type target, bool isAddress)
                     Debug.Assert(!isAddress);
                     Type argType = source.GetElementType();
                     Ldind(il, argType);
-                    Convert(il, argType, target, isAddress);
+                    Convert(il, argType, target, isAddress); //??? always false
```
Comment (serkantkaraca) [inline src/Microsoft.Azure.EventHubs/Amqp/Management/RealProxy.cs]: >//??? always false [](start = 61, length = 18)

Let's remove isAddress then. Another legacy from .NET client.

**Comment by serkantkaraca** [comment]
:shipit:

**Comment by SeanFeldman** [comment]
@serkantkaraca probably can't :shipit: until rebased and comments are addressed. Will try today/tomorrow. Still need #14 to be addressed and have tests running locally to have the confidence to change things 😃

File: src/Microsoft.Azure.EventHubs/Amqp/AmqpMessageConverter.cs (hunk 1 above)
Comment (SeanFeldman) [inline src/Microsoft.Azure.EventHubs/Amqp/AmqpMessageConverter.cs]: done

File: src/Microsoft.Azure.EventHubs/Amqp/Management/RealProxy.cs (hunk 2 above)
Comment (SeanFeldman) [inline src/Microsoft.Azure.EventHubs/Amqp/Management/RealProxy.cs]: done

File: src/Microsoft.Azure.EventHubs/Amqp/AmqpEventHubClient.cs [hunk 3]
```diff
@@ -106,16 +106,13 @@ protected override Task OnCloseAsync()
                     this.ConnectionStringBuilder.Endpoint.AbsoluteUri, // audience
                     this.ConnectionStringBuilder.Endpoint.AbsoluteUri, // endpointUri
                     requiredClaims,
-                    isClientToken,
+                    isClientToken, // ??? always false
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.EventHubs/Amqp/AmqpEventHubClient.cs]: done

**Comment by SeanFeldman** [comment]
@serkantkaraca I've done the following:
- Rebased on the recent master. Tests are green. One test is a bit flaky and fails from time to time.
- Incorporated feedback.
- Added a few cleanups for tests to demonstrate what I'm talking about in #19.

Ready for review/merge.

**Pull request merged**